app/config/EADConfig.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package config
+
+import features.Feature.Feature
+import features.{Feature, FeatureStatus}
+import features.FeatureStatus.FeatureStatus
+import javax.inject.{Inject, Singleton}
+import play.api.Configuration
+
+@Singleton
+class EADConfig @Inject()(configuration: Configuration) {
+
+  private def loadConfig(key: String): String =
+    configuration.getOptional[String](key).getOrElse(throw new Exception(s"Missing configuration key: $key"))
+
+  private def feature2Key(feature: Feature): String = s"microservice.services.features.$feature"
+
+  private def str2FeatureStatus(str: String): FeatureStatus = FeatureStatus.withName(str)
+
+  lazy val defaultFeatureStatus: features.FeatureStatus.Value =
+    FeatureStatus.withName(loadConfig(feature2Key(Feature.default)))
+
+  def featureStatus(feature: Feature): FeatureStatus =
```

Comment:
You probably want it to be private. Otherwise you could check any feature using EADConfig class.

Reply:
Good point, changed